akka-javasdk/src/main/java/akka/javasdk/agent/ConversationMemory.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.agent;
+
+import akka.Done;
+import akka.javasdk.agent.ConversationMemory.Event;
+import akka.javasdk.annotations.TypeName;
+import akka.javasdk.eventsourcedentity.EventSourcedEntity;
+import akka.javasdk.annotations.ComponentId;
+import akka.javasdk.impl.agent.ConversationHistoryImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import static akka.Done.done;
+
+/**
+ * ConversationMemory is an EventSourcedEntity that maintains a limited history of conversation
+ * messages in a FIFO (First In, First Out) style.
+ * <p>
+ * The maximum number of entries in the history can be set dynamically with command setLimitedWindow.
+ * {@link akka.javasdk.client.ComponentClient} can be used to interact directly with this entity.
+ */
+@ComponentId("akka-conversation-memory")
+public final class ConversationMemory extends EventSourcedEntity<ConversationHistory, Event> {
```

Comment:
I wonder if there is a better way to still hide the implementation.. Maybe only doing the conversation from `ConversationHistoryImpl` to `ConversationHistory` on `getHistory` instead? But then the Impl specific one will be on `emptyState` and here..